### DISCUSSION TITLE
point distance

### DISCUSSION THREAD

**Discussion by vdkxmx** [OpenDrift/opendrift#1258] [Q&A]
Hello,i want ask the function "distance_along_trajectory(lon,lat)" is to caculate what distance? is the distance between initial point and ending point?or other distance? And the lon,lat in the bracket is mean what? Very thanks!

**Comment by knutfrode** [comment]
Hi,

This provides the distances between one position and the next along the trajectory given by arrays lon and lat:
https://opendrift.github.io/_modules/opendrift/models/physics_methods.html#distance_along_trajectory

To get the cumulative distance from the start position, you can apply `cumsum()` on the returned array of individual distances.

From a simulation object you can get the total length, individual distances and the particle speed for each trajectory with `o.get_trajectory_lengths()`
https://opendrift.github.io/_modules/opendrift/models/basemodel.html#OpenDriftSimulation.get_trajectory_lengths

**Reply by vdkxmx** [reply]
Thanks for replying, your advise very helpful for me! But now I have another queistion, or idea? I want to run some continuely simulations, i dont know how to describe, just like when a simulation has run to end, finished, then using this finished sinulation as the "start" for next sinulation, because my device cannot sinulate for a long time, like 30 or more years. So would you please give me some advice?

**Comment by knutfrode** [comment]
You could save output of first simulation to a netCDF-file, and then import this with OpenDrift or Xarray, and then read end-positions and use as start-positions for new simulation. But all other things will then be reset, so you may need to tweak end results/files to create a continuous timeseries.
